### 1. 데이터 가져오기

In [1]:
import pandas as pd
lpg_buffer=pd.read_csv("C:/Users/Kim Won Gyeung/Desktop/공모전/2021 빅데이터 분석 및 아이디어 공모전/2.사용_데이터(정제데이터)/1.QGIS에 사용할 데이터(gid 가능데이터)/gid포함 csv파일/수소차버퍼, lpg버퍼 격자데이터/lpg버퍼격자.csv",encoding='cp949')

In [8]:
lpg_buffer

소재지   Latitude   Longitude       gid
0      대전 대덕구 상서동 441-3  36.422211  127.419109  다바928260
1      대전 대덕구 상서동 441-3  36.422211  127.419109  다바931257
2      대전 대덕구 상서동 441-3  36.422211  127.419109  다바932267
3      대전 대덕구 상서동 441-3  36.422211  127.419109  다바919249
4      대전 대덕구 상서동 441-3  36.422211  127.419109  다바928262
...                 ...        ...         ...       ...
48829  대전 중구 중촌동 413-43  36.347966  127.405889  다바913167
48830  대전 중구 중촌동 413-43  36.347966  127.405889  다바929174
48831  대전 중구 중촌동 413-43  36.347966  127.405889  다바916179
48832  대전 중구 중촌동 413-43  36.347966  127.405889  다바914164
48833  대전 중구 중촌동 413-43  36.347966  127.405889  다바912174

[48834 rows x 4 columns]

In [3]:
df=pd.read_csv("C:/Users/Kim Won Gyeung/Desktop/공모전/2021 빅데이터 분석 및 아이디어 공모전/2.사용_데이터(정제데이터)/4. 선형회귀 돌릴 데이터/수소차종속변수_선형회귀.csv",encoding='cp949')

In [5]:
df.head()

gid  수소충전소  LPG충전소  차량등록수  생산가능인구수  교통노드  건물연면적  관광상권  부동산상권  생활서비스상권  \
0  다마846996      0       0      0        0     0    0.0     0      0        0   
1  다마847995      0       0      0        0     0    0.0     0      0        0   
2  다마847996      0       0      0        0     0    0.0     0      0        0   
3  다마847997      0       0      0        0     0    0.0     0      0        0   
4  다마847998      0       0      0        0     0    0.0     0      0        0   

   소매상권  숙박상권  스포츠상권  음식상권  학문상권  유치원초중고  화재발생건수  교통량  종속변수_수소차  
0     0     0      0     0     0       0     0.0  0.0       0.0  
1     0     0      0     0     0       0     0.0  0.0       0.0  
2     0     0      0     0     0       0     0.0  0.0       0.0  
3     0     0      0     0     0       0     0.0  0.0       0.0  
4     0     0      0     0     0       0     0.0  0.0       0.0

### 2. 버퍼에 데이터 붙이기

In [9]:
gid1 = lpg_buffer.merge(df,on='gid',how='inner')

In [50]:
gid1

소재지   Latitude   Longitude       gid  수소충전소  LPG충전소  \
0       대전 대덕구 상서동 441-3  36.422211  127.419109  다바928260      0       0   
1      대전 대덕구 상서동 349-12  36.425293  127.420454  다바928260      0       0   
2         대전 대덕구 상서동 440  36.422242  127.419851  다바928260      0       0   
3       대전 대덕구 평촌동 226-5  36.436035  127.425928  다바928260      0       0   
4      대전 대덕구 상서동 236-32  36.428133  127.418259  다바928260      0       0   
...                  ...        ...         ...       ...    ...     ...   
48829     대전 중구 석교동 17-2  36.307001  127.443016  다바935111      0       0   
48830     대전 중구 석교동 17-2  36.307001  127.443016  다바957128      0       0   
48831     대전 중구 석교동 17-2  36.307001  127.443016  다바953127      0       0   
48832     대전 중구 석교동 17-2  36.307001  127.443016  다바947105      0       0   
48833     대전 중구 석교동 17-2  36.307001  127.443016  다바957129      0       0   

       차량등록수  생산가능인구수  교통노드    건물연면적  ...  생활서비스상권  소매상권  숙박상권  스포츠상권  음식상권  \
0         40       65     2  1487.49  ...        0     0     0      0     0   
1         40       65     2  1487.49  ...        0     0     0      0     0   
2         40       65     2  1487.49  ...        0     0     0      0     0   
3         40       65     2  1487.49  ...        0     0     0      0     0   
4         40       65     2  1487.49  ...        0     0     0      0     0   
...      ...      ...   ...      ...  ...      ...   ...   ...    ...   ...   
48829      0        0     0     0.00  ...        0     0     0      0     0   
48830      0        0     0     0.00  ...        0     0     0      0     0   
48831      0        0     0     0.00  ...        0     0     0      0     0   
48832      0        0     0     0.00  ...        0     0     0      0     0   
48833      0        0     0     0.00  ...        0     0     0      0     0   

       학문상권  유치원초중고  화재발생건수  교통량  종속변수_수소차  
0         0       0     0.0  1.0       1.0  
1         0       0     0.0  1.0       1.0  
2         0       0     0.0  1.0       1.0  
3         0       0     0.0  1.0       1.0  
4         0       0     0.0  1.0       1.0  
...     ...     ...     ...  ...       ...  
48829     0       0     0.0  0.0       0.0  
48830     0       0     0.0  0.0       0.0  
48831     0       0     0.0  0.0       0.0  
48832     0       0     0.0  0.0       0.0  
48833     0       0     0.0  0.0       0.0  

[48834 rows x 22 columns]

In [21]:
address=gid1['소재지'].unique()
address

array(['대전 대덕구 상서동 441-3', '대전 대덕구 상서동 349-12', '대전 대덕구 상서동 440',
       '대전 대덕구 평촌동 226-5', '대전 대덕구 상서동 236-32', '대전 대덕구 신탄진동 294-5',
       '대전 대덕구 신일동 1684-1', '대전 대덕구 신대동 433-2', '대전 대덕구 신대동 426',
       '대전 대덕구 연축동 294-2', '대전 대덕구 대화동 16-140', '대전 대덕구 대화동 3-1',
       '대전 대덕구 읍내동 511', '대전 대덕구 오정동 432-1', '대전 대덕구 오정동 478-4',
       '대전 동구 성남2동 502-4', '대전 동구 가양동 422-7', '대전 대덕구 대화동 411-5',
       '대전 대덕구 대화동 549-16', '대전 중구 중촌동 193-1', '대전 중구 중촌동 194-1',
       '대전 중구 중촌동 198-1', '대전 중구 중촌동 413-43', '대전 중구 오류동 190-18',
       '대전 중구 오류동 182-1', '대전 서구 가장동 31-4', '대전 동구 구도동 366',
       '대전 동구 대성동 266', '대전 동구 대성동 206-1', '대전 동구 삼정동 118-11',
       '대전 중구 석교동 17-2', '대전 서구 변동 12-7', '대전 중구 산성동 166-1',
       '대전 서구 정림동 8-57', '대전 서구 정림동 293-10', '대전 서구 월평1동 105-2',
       '대전 중구 안영동 280', '대전 유성구 장대동 220-18', '대전 유성구 학하동 501-2',
       '대전 유성구 원내동 427', '대전 유성구 교촌동 331-1', '대전 유성구 방동 474-4',
       '대전 유성구 갑동 404', '대전 유성구 갑동 398', '대전 유성구 덕진동 50',
       '대전 유성구 송정동 300-1'], dtype

In [53]:
for index,i in enumerate(address):
    globals()['LPG_{}'.format(index)] = gid1[gid1['소재지']==i].iloc[:,:-1]

### 3. 각 LPG버퍼의 평균값 생성

각 변수 별 기울기 : 
[-0.01282657  0.01511532  0.01040095  0.00572691  0.85205971 -0.55621325
   0.08490633 -0.10579936 -0.54213626  0.62621851 -0.03215032 -0.45079255
  -0.65436459 -0.1676087  -1.02624619]

In [40]:
coef=[-0.01282657,  0.01511532,  0.01040095,  0.00572691,  0.85205971, -0.55621325,
   0.08490633, -0.10579936, -0.54213626 , 0.62621851, -0.03215032 ,-0.45079255,
  -0.65436459, -0.1676087,  -1.02624619]

In [66]:
import numpy as np
score_df = {'address':address,
           'score':np.zeros(46)}

In [68]:
score = pd.DataFrame(score_df)
score

address  score
0    대전 대덕구 상서동 441-3    0.0
1   대전 대덕구 상서동 349-12    0.0
2      대전 대덕구 상서동 440    0.0
3    대전 대덕구 평촌동 226-5    0.0
4   대전 대덕구 상서동 236-32    0.0
5   대전 대덕구 신탄진동 294-5    0.0
6   대전 대덕구 신일동 1684-1    0.0
7    대전 대덕구 신대동 433-2    0.0
8      대전 대덕구 신대동 426    0.0
9    대전 대덕구 연축동 294-2    0.0
10  대전 대덕구 대화동 16-140    0.0
11     대전 대덕구 대화동 3-1    0.0
12     대전 대덕구 읍내동 511    0.0
13   대전 대덕구 오정동 432-1    0.0
14   대전 대덕구 오정동 478-4    0.0
15   대전 동구 성남2동 502-4    0.0
16    대전 동구 가양동 422-7    0.0
17   대전 대덕구 대화동 411-5    0.0
18  대전 대덕구 대화동 549-16    0.0
19    대전 중구 중촌동 193-1    0.0
20    대전 중구 중촌동 194-1    0.0
21    대전 중구 중촌동 198-1    0.0
22   대전 중구 중촌동 413-43    0.0
23   대전 중구 오류동 190-18    0.0
24    대전 중구 오류동 182-1    0.0
25     대전 서구 가장동 31-4    0.0
26      대전 동구 구도동 366    0.0
27      대전 동구 대성동 266    0.0
28    대전 동구 대성동 206-1    0.0
29   대전 동구 삼정동 118-11    0.0
30     대전 중구 석교동 17-2    0.0
31      대전 서구 변동 12-7    0.0
32    대전 중구 산성동 166-1    0.0
33     대전 서구 정림동 8-57    0.0
34   대전 서구 정림동 293-10    0.0
35   대전 서구 월평1동 105-2    0.0
36      대전 중구 안영동 280    0.0
37  대전 유성구 장대동 220-18    0.0
38   대전 유성구 학하동 501-2    0.0
39     대전 유성구 원내동 427    0.0
40   대전 유성구 교촌동 331-1    0.0
41    대전 유성구 방동 474-4    0.0
42      대전 유성구 갑동 404    0.0
43      대전 유성구 갑동 398    0.0
44      대전 유성구 덕진동 50    0.0
45   대전 유성구 송정동 300-1    0.0

In [69]:
for i in range(46):
    score['score'][i]=sum(globals()['LPG_{}'.format(i)].mean()[4:]*coef)

<ipython-input-69-cd3cec7ffc6d>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  score['score'][i]=sum(globals()['LPG_{}'.format(i)].mean()[4:]*coef)


In [77]:
score.sort_values('score',ascending=False)

address      score
18  대전 대덕구 대화동 549-16  13.458009
21    대전 중구 중촌동 198-1  11.777405
23   대전 중구 오류동 190-18  10.691640
24    대전 중구 오류동 182-1  10.537839
22   대전 중구 중촌동 413-43  10.401629
35   대전 서구 월평1동 105-2   9.761938
20    대전 중구 중촌동 194-1   9.493043
19    대전 중구 중촌동 193-1   9.405701
31      대전 서구 변동 12-7   8.662368
25     대전 서구 가장동 31-4   8.607049
17   대전 대덕구 대화동 411-5   8.388323
37  대전 유성구 장대동 220-18   7.024080
32    대전 중구 산성동 166-1   6.943762
6   대전 대덕구 신일동 1684-1   6.595987
10  대전 대덕구 대화동 16-140   5.751227
34   대전 서구 정림동 293-10   5.677523
14   대전 대덕구 오정동 478-4   5.560532
13   대전 대덕구 오정동 432-1   5.504686
11     대전 대덕구 대화동 3-1   5.409913
44      대전 유성구 덕진동 50   5.377564
15   대전 동구 성남2동 502-4   5.218322
12     대전 대덕구 읍내동 511   5.184663
16    대전 동구 가양동 422-7   5.084985
33     대전 서구 정림동 8-57   4.569957
5   대전 대덕구 신탄진동 294-5   4.326902
7    대전 대덕구 신대동 433-2   4.227235
30     대전 중구 석교동 17-2   4.201701
8      대전 대덕구 신대동 426   4.169140
39     대전 유성구 원내동 427   3.773287
3    대전 대덕구 평촌동 226-5   3.250895
40   대전 유성구 교촌동 331-1   3.192075
9    대전 대덕구 연축동 294-2   3.161824
4   대전 대덕구 상서동 236-32   2.823709
28    대전 동구 대성동 206-1   2.730294
29   대전 동구 삼정동 118-11   2.551749
1   대전 대덕구 상서동 349-12   2.195009
27      대전 동구 대성동 266   2.036550
0    대전 대덕구 상서동 441-3   2.027150
2      대전 대덕구 상서동 440   1.983130
38   대전 유성구 학하동 501-2   1.951998
43      대전 유성구 갑동 398   0.982839
42      대전 유성구 갑동 404   0.686715
36      대전 중구 안영동 280   0.632229
26      대전 동구 구도동 366   0.375149
45   대전 유성구 송정동 300-1  -0.045503
41    대전 유성구 방동 474-4  -0.125841

### 결론
대전 대덕구 대화동 549-16번지가 13.458009점으로 제일 점수가 높아
수소차 충전소 입지에 제일 탁월하다.